lib/Serialization/Serialization.cpp

```diff
@@ -1286,16 +1197,8 @@ void Serializer::writeGenericSignature(const GenericSignature *sig) {
   writeGenericRequirements(sig->getRequirements(), DeclTypeAbbrCodes);
 }
 
-void Serializer::writeGenericEnvironment(const GenericEnvironment *env) {
-  using namespace decls_block;
-
-  // Record the offset of this generic environment.
-  auto id = GenericEnvironmentIDs[env];
-  assert(id != 0 && "generic environment not referenced properly");
-  (void)id;
-
-  assert((id - 1) == GenericEnvironmentOffsets.size());
-  assert(env != nullptr);
+void Serializer::writeNextGenericEnvironment() {
+  const GenericEnvironment *env = GenericEnvironmentsToSerialize.peekNext();
 
   // Determine whether we must use SIL mode, because one of the generic
```

Comment:
Not part of your change, but the loop here to set `SILMode` could be made to use `std::any_of` if you feel like it.

Reply:
Heh. I started down that path (using `llvm::any_of` even) and then decided that I didn't want to mix it in with this change. I might do it as a follow-up.